src/Sarif.Multitool/TransformCommand.cs

```diff
@@ -67,7 +67,12 @@ public int Run(TransformOptions transformOptions)
                     else
                     {
                         // We have a pre-release v2 file that we should upgrade to current. 
-                        string sarifText = PrereleaseCompatibilityTransformer.UpdateToCurrentVersion(_fileSystem.ReadAllText(inputFilePath), formatting: formatting);
+                        PrereleaseCompatibilityTransformer.UpdateToCurrentVersion(
```

Comment:
>PrereleaseCompatibilityTransformer [](start = 24, length = 34)

This helper now returns the transformed contents as a SarifLog. We need to do this in the transformer in order to invoke the visitors that rewrite indices. There is no sense converting this log file back to text again, so that a consumer deserializes yet one more time. Instead, we have a rather cumbersome API that can be used to get the transformed text as an out arg or the transformed log file as a returned value. Note that the out argument is set to the original contents that are passed to the API, in failure cases and when the file doesn't need to be touched at all.

Reply:
IMO the out parameter should start out as `null` to handle the failure case. Later on, if no transform was necessary, it can be set to the input text.

---
In reply to: [243701919](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243701919) [](ancestors = 243701919)

Reply:
I looked at `PrereleaseCompatibilityTransformer.UpdateToCurrentVersion` and contrary to this comment, it _does_ initially set the out parameter to `null`, as I suggested it should, so there's nothing to do here.

---
In reply to: [243852926](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243852926) [](ancestors = 243852926,243701919)